diff --git a/femtools/Solvers.F90 b/femtools/Solvers.F90
index 23ee87faf..3a254822c 100644
--- a/femtools/Solvers.F90
+++ b/femtools/Solvers.F90
@@ -1926,12 +1926,6 @@ subroutine create_ksp_from_options(ksp, mat, pmat, solver_option_path, parallel,
       ! need to call this before the subpc can be retrieved:
       call PCSetup(pc, ierr)
       
-      if (pctype==PCBJACOBI) then
-        call PCBJACOBIGetSubKSP(pc, PETSC_NULL_INTEGER, PETSC_NULL_INTEGER, subksp, ierr)
-      else
-        call PCASMGetSubKSP(pc, PETSC_NULL_INTEGER, PETSC_NULL_INTEGER, subksp, ierr)
-      end if
-
       call KSPGetPC(subksp, subpc, ierr)
       ! recursively call to setup the subpc
       ewrite(2,*) "Going into setup_pc_from_options for the subpc within the local domain."
@@ -1952,7 +1946,6 @@ subroutine create_ksp_from_options(ksp, mat, pmat, solver_option_path, parallel,
        call PCSetType(pc, PCBJACOBI, ierr)
        ! need to call this before the subpc can be retrieved:
        call PCSetup(pc, ierr)
-       call PCBJACOBIGetSubKSP(pc, PETSC_NULL_INTEGER, PETSC_NULL_INTEGER, subksp, ierr)
        call KSPGetPC(subksp, subpc, ierr)
        call PCSetType(subpc, pctype, ierr)
